### PR TITLE
Fix `owasp-protection-global-unsafe` rule failing to detect previously recognised cases

### LINKED ISSUE #431: `owasp-protection-global-unsafe` rule is failing to detect previously recognised cases
I'm not sure what's going on here but in the example appspec, where security has been removed from the `DELETE /pets/{petId}` endpoint, the `owasp-protection-global-unsafe` doesn't pick it up any more:

```yaml
openapi: 3.0.1
info:
  title: Security disabled for specific endpoint
  version: "0.1"
servers:
  - url: https://example.com
paths:
  /pets:
    post:
      summary: "create a pet"
      responses:
        "200":
          description: some descriptive text
  /pets/{petId}:
    parameters:
      - name: petId
        in: path
        required: true
        schema:
          type: integer
    get:
      summary: "get a pet"
      responses:
        "200":
          description: some descriptive text
    delete:
      summary: "delete a pet"
      responses:
        "200":
          description: some descriptive text
      security: []
components:
  securitySchemes:
    jwt:
      type: http
      scheme: bearer
      bearerFormat: JWT
security:
  - jwt: []
```

Relevant part of [the OpenAPI spec](https://swagger.io/specification/), under the Operation Object see the description for the `security` field:

> To remove a top-level security declaration, an empty array can be used.

This was detected in earlier versions of vacuum.

I'm guessing there's something funky in the logic here:

https://github.com/daveshanley/vacuum/blob/0e30ba4eacac7b5694e3c65cc23f79a5a5394b7a/functions/owasp/check_security.go#L104-L105

When I've played around with it it starts erroneously reporting that security is also missing from the `POST /pets` endpoint too though.

### PR DESCRIPTION
**Pull request by TheTeaCat**
Fixes #431 

Blocked by https://github.com/pb33f/doctor/pull/1 - once merged I can remove the replacement from the `go.mod` here.

### CODE REVIEW

**Comment by daveshanley** [comment]
The PR for the Doctor has been merged and sits at `v0.0.4`

**Comment by TheTeaCat** [comment]
@daveshanley ready for review here, CI passed

File: functions/owasp/check_security.go [hunk 1]
```diff
@@ -101,8 +101,7 @@ func (cd CheckSecurity) RunRule(nodes []*yaml.Node, context model.RuleFunctionCo
 
 				}
 
-				if len(opValue.Security) <= 0 && globalSecurity != nil &&
-					(globalSecurity[0].Value.Requirements == nil || globalSecurity[0].Value.Requirements.Len() <= 0) {
+				if opValue.Security != nil && len(opValue.Security) <= 0 {
```
Comment (daveshanley) [inline functions/owasp/check_security.go:104]: Is there a reason you removed the check for global security here?

Comment (TheTeaCat) [reply functions/owasp/check_security.go:104]: Oh I think I see it, I'm missing a case for a spec like this?:

```yaml
openapi: 3.0.1
info:
  title: Test spec
  version: 0.0.1
security:
  - {}
paths:
  /insecure:
    get:
      responses:
        default:
          description: Default response
```

Comment (TheTeaCat) [reply functions/owasp/check_security.go:104]: I've added a test case where the second entry in the global security is an empty object, making security optional. I think that's what the `(globalSecurity[0].Value.Requirements == nil || globalSecurity[0].Value.Requirements.Len() <= 0)` was doing, but it was only checking the 0th element?

**Review by daveshanley — APPROVED** [review]
LGTM! Thank you for your contribution.

**Pull request merged**